Set SSH_USER for root commands

scjody opened the pull request:
Previously, SSH_USER was not set for plugin:* or ssh-keys:add, which contradicts the user-auth trigger documentation.

Fixes #2700

* [x] Integration test with `dokku-acl` to make sure this fixes the issue

josegonzalez commented on dokku:
```diff
@@ -58,6 +58,8 @@ fi
 if [[ $(id -un) != "root" && $1 =~ ^plugin:.* && $1 != "plugin:help" && $1 != "plugin:list" ]] || [[ $(id -un) != "root" && $1 == "ssh-keys:add" ]]; then
   dokku_log_fail "This command must be run as root"
+else
+  export SSH_USER=root
```
Doesn't this force-override it always? We should only set the SSH_USER here if its one of the commands that requires root.

scjody commented:
@josegonzalez Sorry, I didn't really think that through. I don't understand how adding an `else` would work in that case, but here's the other approach: move the `export SSH_USER` outside the first `if`. I've checked and this fixes the problem for us.

I also removed the extra `$(id -un)` commands and replaced them with `$SSH_USER` but if you'd rather stick with an explicit `$(id -un)` I'm happy to change them back.

josegonzalez commented:
```
if requires_root
  if ! root
    error
  else
    SSH_USER=derp
fi
```

or

```
if requires_root and not root
  error
fi
export SSH_USER=${SSH_USER:-some-default}
```

scjody commented:
@josegonzalez I implemented your first suggestion. Does this look OK now?
